receiver/dockerstatsreceiver/receiver.go

```diff
@@ -126,6 +126,7 @@ func (r *receiver) recordContainerStats(now pcommon.Timestamp, containerStats *d
 	r.recordMemoryMetrics(now, &containerStats.MemoryStats)
 	r.recordBlkioMetrics(now, &containerStats.BlkioStats)
 	r.recordNetworkMetrics(now, &containerStats.Networks)
+	r.recordPidsMetrics(now, &containerStats.PidsStats)
```

Comment:
Since [PidStats](https://github.com/moby/moby/blob/v23.0.4/api/types/stats.go#L159) is not defined as a pointer in docker types it will be an empty struct, therefore this pointer cannot be nil even if `pids_stats` are not reported.

The stats response in the new test case [pids_no_stats](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21111/commits/7526d761f8709276f48869cd32ab591cf23ff329#diff-26382ae6339e56dd697451b6cc6c37f948cf8712cf30621a74ea1405c531a514R208-R210) does not include `pids_stats` metrics, showing this use-case.